### PR TITLE
chore(weave): Correctly identify all non-digest aliases as impure

### PR DESCRIPTION
**Pull request by tssweeney**
This PR enahances impure identification to more than just latest, including all non-hash aliases.

### CODE REVIEW

**Review by connieelee — APPROVED** [review]

File: weave-js/src/core/client/cachePolicy.ts [hunk 1]
```diff
@@ -24,12 +24,33 @@ const getStringValFromNode = (
   return undefined;
 };
 
+const WANDB_COMMIT_HASH_LENGTH = 20;
+
+const isHexString = (s: string): boolean => {
+  return s.length % 2 === 0 && /^[0-9a-f]+$/i.test(s);
```
Comment (jamie-rasmussen) [inline weave-js/src/core/client/cachePolicy.ts:30]: I'd remove `s.length % 2 === 0` - the function name doesn't suggest it is doing that check, and it is covered by the exact length check in `likelyCommitHash` (could reverse order to do length check before regexp test)

File: weave-js/src/core/client/cachePolicy.ts [hunk 2]
```diff
@@ -24,12 +24,33 @@ const getStringValFromNode = (
...
+  const alias = aliasPart.split('/')[0];
+  return alias;
+};
+
 const nodeIsImpureGetOp = (node: GraphTypes.Node<any>): boolean => {
   if (isGetNode(node)) {
     const uriVal = getStringValFromNode(node.fromOp.inputs.uri);
-    if (uriVal?.includes(':latest')) {
-      return true;
+    if (uriVal) {
```
Comment (jamie-rasmussen) [inline weave-js/src/core/client/cachePolicy.ts:48]: I suggest `if (uriVal && likelyCommitHash(getAliasFromUri(uriVal))) {`

File: weave-js/src/core/client/cachePolicy.ts [hunk 3]
```diff
@@ -24,12 +24,33 @@ const getStringValFromNode = (
   return undefined;
 };
 
+const WANDB_COMMIT_HASH_LENGTH = 20;
+
+const isHexString = (s: string): boolean => {
+  return s.length % 2 === 0 && /^[0-9a-f]+$/i.test(s);
+};
+
+const likelyCommitHash = (version: string): boolean => {
```
Comment (jamie-rasmussen) [inline weave-js/src/core/client/cachePolicy.ts:33]: Maybe continue with the `is` prefixed naming for a boolean result?

**Review by jamie-rasmussen — APPROVED** [review]

File: weave-js/src/core/client/cachePolicy.ts (hunk 1 above)
Comment (tssweeney) [reply weave-js/src/core/client/cachePolicy.ts:30]: 👍

File: weave-js/src/core/client/cachePolicy.ts (hunk 3 above)
Comment (tssweeney) [reply weave-js/src/core/client/cachePolicy.ts:33]: 👍

File: weave-js/src/core/client/cachePolicy.ts (hunk 2 above)
Comment (tssweeney) [reply weave-js/src/core/client/cachePolicy.ts:48]: 👍

**Pull request merged**